operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/InformerEventSource.java

```diff
@@ -74,28 +75,33 @@
   private final EventRecorder<R> eventRecorder = new EventRecorder<>();
   // we need direct control for the indexer to propagate the just update resource also to the index
   private final PrimaryToSecondaryIndex<R> primaryToSecondaryIndex;
+  private final PrimaryToSecondaryMapper<P> primaryToSecondaryMapper;
 
   public InformerEventSource(
       InformerConfiguration<R> configuration, EventSourceContext<P> context) {
-    super(context.getClient().resources(configuration.getResourceClass()), configuration);
-    this.configuration = configuration;
-    primaryToSecondaryIndex =
-        new PrimaryToSecondaryIndex<>(configuration.getSecondaryToPrimaryMapper());
+    this(configuration, context.getClient());
   }
 
   public InformerEventSource(InformerConfiguration<R> configuration, KubernetesClient client) {
     super(client.resources(configuration.getResourceClass()), configuration);
     this.configuration = configuration;
-    primaryToSecondaryIndex =
-        new PrimaryToSecondaryIndex<>(configuration.getSecondaryToPrimaryMapper());
+    primaryToSecondaryMapper = configuration.getPrimaryToSecondaryMapper();
+    if (primaryToSecondaryMapper == null) {
+      primaryToSecondaryIndex =
+          new PrimaryToSecondaryIndex<>(configuration.getSecondaryToPrimaryMapper());
+    } else {
+      primaryToSecondaryIndex = null;
```

Comment:
Maybe it'd be cleaner to have a no-op index instead of setting it to `null` and having to guard for `null` everywhere else?

Reply:
Not sure about that, it is very explicit this way, that the index is not used in that case.

Reply:
But also fine with that. Will try how that looks.

Reply:
Hmm, here it needs to be explicit:
https://github.com/java-operator-sdk/java-operator-sdk/blob/b7720f974ad2c450c9df6ac5297847af150b33d1/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/InformerEventSource.java#L193-L193

Reply:
One doesn't prevent the other: we could have a NOOP index and still be able to make it explicit when needed (i.e. `useSecondaryToPrimaryIndex` could check if the set index is the NOOP one instead of `null` where we need to know for sure whether to use it or not). Not a showstopper in any case.

Reply:
Changed.